browser/search/ntp_utils.cc

```diff
@@ -5,32 +5,96 @@
 
 #include "brave/browser/search/ntp_utils.h"
 
+#include "base/logging.h"
 #include "brave/browser/profiles/profile_util.h"
 #include "brave/common/pref_names.h"
+#include "brave/components/crypto_dot_com/browser/buildflags/buildflags.h"
 #include "chrome/common/pref_names.h"
 #include "components/pref_registry/pref_registry_syncable.h"
 #include "components/prefs/pref_registry_simple.h"
 #include "components/prefs/pref_service.h"
 
+#if BUILDFLAG(CRYPTO_DOT_COM_ENABLED)
+#include "brave/components/crypto_dot_com/common/pref_names.h"
+#endif
+
 namespace {
 
 void ClearNewTabPageProfilePrefs(Profile* profile) {
   PrefService* prefs = profile->GetPrefs();
   prefs->ClearPref(kNewTabPageShowTopSites);
 }
 
+const char* const widget_pref_names[] = {
+  kNewTabPageShowRewards,
+  kNewTabPageShowTogether,
+  kNewTabPageShowBinance,
+#if BUILDFLAG(CRYPTO_DOT_COM_ENABLED)
+  kCryptoDotComNewTabPageShowCryptoDotCom,
+#endif
+  kNewTabPageShowGemini
+};
+
 }  // namespace
 
 namespace new_tab_page {
 
 void MigrateNewTabPagePrefs(Profile* profile) {
   // Migrate over to the Chromium setting for shortcuts visible
   // Only sets the value if user has changed it
-  const PrefService::Preference* pref =
-      profile->GetPrefs()->FindPreference(kNewTabPageShowTopSites);
-  if (pref->HasUserSetting()) {
-    profile->GetPrefs()->SetBoolean(prefs::kNtpShortcutsVisible,
-      profile->GetPrefs()->GetBoolean(kNewTabPageShowTopSites));
+  auto* prefs = profile->GetPrefs();
+  const PrefService::Preference* top_sites_pref =
+      prefs->FindPreference(kNewTabPageShowTopSites);
+  if (top_sites_pref->HasUserSetting()) {
+    prefs->SetBoolean(prefs::kNtpShortcutsVisible,
+      prefs->GetBoolean(kNewTabPageShowTopSites));
+  }
+
+  // The toggle to turn off all widgets used to simply turn off
+  // all existing widgets. We later introduced a pref so that future
+  // new widgets do not show for that user. Perform a one-off migration
+  // for known widgets at the time to set this new pref.
+  // Clear and remove the migration pref at some point in the future.
+  // Note(petemill): It was created 2021-06-09.
+  const PrefService::Preference* migrate_hide_widgets_pref =
+      prefs->FindPreference(kNewTabPageMigratedHideAllWidgetsPref);
+  if (!migrate_hide_widgets_pref->HasUserSetting()) {
```

Comment:
IMO, we don't need to add `kNewTabPageMigratedHideAllWidgetsPref`.
Instead, setting false to `kNewTabPageHideAllWidgets` explicitely. WDYT?

Reply:
makes sense

Reply:
fixed